frontend/src/todos/todosList.tsx

```diff
@@ -104,57 +104,65 @@ function TodosList() {
       setTodos({ status: 'failed', error: error.message });
     }
   }
+
+  function Spinner(){
+    return(
+      <div
+      style={{
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        zIndex: 1000,
+      }}
+    >
+      <RingLoader color=" #36ccae" size={150} />
+    </div>
+    )
+  }
+
+  function Body(prpos: {isPending:boolean, todos: Todo[]}){
```

Comment:
extract form the component and also give it a better name (you can call it TodoListPresenter, and the current component TodoListContainer - it's kind of convention we have for this kind of stuffs)